### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by julianxhokaxhiu**
**Walkmesh:**
- Draw background if present
- Reset state on window close

**FieldModel:**
- Add missing update event call
- Reset state on window close

### CODE REVIEW

**Comment by TurBoss** [comment]
Hello,

cmake does not complain about missing Qt5Zlib

```
➜  build git:(develop) ✗ cmake ..
-- The C compiler identification is GNU 10.2.0
-- The CXX compiler identification is GNU 10.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /home/turboss/Projetcs/makoureactor/build
➜  build git:(develop) ✗ make
Scanning dependencies of target Makou_Reactor
[ 11%] Creating directories for 'Makou_Reactor'
[ 22%] No download step for 'Makou_Reactor'
[ 33%] Performing reconfigure step for 'Makou_Reactor'
Forcing a superbuild reconfigure
[ 44%] No update step for 'Makou_Reactor'
[ 55%] No patch step for 'Makou_Reactor'
[ 66%] Performing configure step for 'Makou_Reactor'
-- The C compiler identification is GNU 10.2.0
-- The CXX compiler identification is GNU 10.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Qt found: 
CMake Error at /lib/cmake/Qt5/Qt5Config.cmake:28 (find_package):
  Could not find a package configuration file provided by "Qt5Zlib" with any
  of the following names:

    Qt5ZlibConfig.cmake
    qt5zlib-config.cmake

  Add the installation prefix of "Qt5Zlib" to CMAKE_PREFIX_PATH or set
  "Qt5Zlib_DIR" to a directory containing one of the above files.  If
  "Qt5Zlib" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  src/CMakeLists.txt:76 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/turboss/Projetcs/makoureactor/build/ep/Build/Makou_Reactor/CMakeFiles/CMakeOutput.log".
make[2]: *** [CMakeFiles/Makou_Reactor.dir/build.make:127: ep/Stamp/Makou_Reactor/Makou_Reactor-configure] Error 1
make[1]: *** [CMakeFiles/Makefile2:115: CMakeFiles/Makou_Reactor.dir/all] Error 2
make: *** [Makefile:103: all] Error 2
```

thanks

**Comment by julianxhokaxhiu** [comment]
Hi @TurBoss,

The new CMake build is not yet compatible with Linux or Mac, we need to fix the QT detection. But if you have a patch regarding this feel free to open an PR :)

Until then the build works perfectly under Windows.

Cheers!

File: src/widgets/WalkmeshWidget.cpp [hunk 1]
```diff
@@ -354,6 +358,51 @@ void WalkmeshWidget::drawLine(const Vertex_sr &pointA, const Vertex_sr &pointB,
...
+					{1.0f, 1.0f},
+				},
+				{
+					{1.0f, 1.0f, 1.0f, 1.0f},
+					{1.0f, 1.0f, 1.0f, 1.0f},
+					{1.0f, 0.0f},
+				}
+    };
+
+    uint32_t indices[] = {
```
Comment (myst6re) [inline src/widgets/WalkmeshWidget.cpp:388]: Inconsistent indentations

**Review by myst6re — CHANGES_REQUESTED** [review]

File: src/widgets/WalkmeshWidget.cpp (hunk 1 above)
Comment (julianxhokaxhiu) [reply src/widgets/WalkmeshWidget.cpp:388]: Fixed!

**Pull request merged**